chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt

```diff
@@ -170,6 +172,53 @@ class RoomSpek : Spek({
 
             assertThat(badEvents).isEmpty()
         }
+
+        it("always has the correct memberUserIds after room data is changed") {
+
+            setUpInstanceWith(createDefaultRole(), newUsers(PUSHERINO, ALICE), newRoom(GENERAL, ALICE))
+            val superUser = chatFor(SUPER_USER).connect().assumeSuccess()
+            val roomUpdated = CountDownLatch(1)
+
+            superUser.subscribeToRoomMultipart(superUser.generalRoom) { event ->
+                when (event) {
+                    is RoomEvent.RoomUpdated -> {
+                        roomUpdated.countDown()
+                        assertThat(event.room.memberUserIds.size).isEqualTo(2)
```

Comment:
Because this assertion is an a callback, we should do it before decrementing the latch, otherwise there is a chance that the test will continue and finish before the assertion is run (in which case it might fail and not be reported).

Reply:
Good point. I hadn't considered that! thank you.